File: Blind 75 LeetCode/Chahat Rajput/Unique Paths/unique_paths.cpp
class Solution {
public:
    int t[101][101];
    int solve(int i,int j,int m,int n){
        if(i== m-1 && j== n-1){
            return 1;
        }
        if(i<0 || i>=m || j<0 || j>=n){
            return 0;
        }

        if(t[i][j] != -1){
            return t[i][j];
        }

        int right=solve(i,j+1,m,n);
        int down=solve(i+1,j,m,n);

        return t[i][j]=right+down;
    }
    int uniquePaths(int m, int n) {
        memset(t,-1,sizeof(t));
        return solve(0,0,m,n);
    }
};

File: Blind 75 LeetCode/Subhangi dhasmana/Unique paths/Unique paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp = [[0 for _ in range(n)] for _ in range(m)]
        dp[0][0] = 1

        for i in range(m):
            for j in range(n):
                val1 = 0
                val2 = 0
                if(j-1 >= 0):
                    val1 = dp[i][j-1]
                if(i-1 >= 0):
                    val2 = dp[i-1][j]
                dp[i][j] = dp[i][j] + val1 + val2
    

        return dp[m-1][n-1]